lib/saml_authenticator.rb

```diff
@@ -113,6 +113,33 @@ def after_authenticate(auth)
       username ||= uid
       username
     end
+    
+    if attributes.present?
+        
+        email_extrafield = GlobalSetting.try(:saml_extrafield_email).presence || ""
+        firstname_extrafield = GlobalSetting.try(:saml_extrafield_firstname).presence || ""
+        lastname_extrafield = GlobalSetting.try(:saml_extrafield_lastname).presence || ""
+        company_extrafield = GlobalSetting.try(:saml_extrafield_company).presence || ""
+        
+        unless email_extrafield.nil? || email_extrafield.empty?
```

Comment:
You could (and should!) use `field.blank?` for these checks

Reply:
You right, i'm a ruby newbie... Here is the changes.